clconf/clconf.go

```diff
@@ -110,8 +111,8 @@ func GetValue(conf interface{}, keyPath string) (interface{}, error) {
 }
 
 // LoadConf will load all configurations provided.  In order of precedence
-// (highest last), files, overrides.
-func LoadConf(files []string, overrides []string) (map[interface{}]interface{}, error) {
+// (highest last), files, overrides, stream.
```

Comment:
Honestly I think the env vars should have the highest precedence but given the current design I couldn't make that work. As far as the interplay between the rest? I'm not sure it matters as long as it's known. Could go a step further and have an option to dictate the priority...

Reply:
I think env vars should have lowest priority... surprise we disagree.  I cant think of any other situation in which an environment variable outweighs an equivalent command line option.  In every situation I can think of env vars are a fallback.

Dictating priority is possible, but not sure there is a use case yet that would make the implementation worth the effort though.

Changing the order of the current is obviously not an option as it would effect the established contract...  So i guess the only question is before/between/after yamls/overrides.  After is as good as any i guess, so may as well keep it.  If we decide to allow for specified priority later, then we can do that...

Reply:
My thought with the env vars is that they're easily controllable at runtime to debug something without disturbing other parts of the deployment (volumes, secrets, configmaps, etc). If we wanted to (e.g.) alter a single value we can `kubectl set env` and add an extra value to override any files/mounts/etc then just bounce the pod without having to edit any _real_ config.

Reply:
Nothing stops that...  In containers you do not specify command line options, and as such, env vars win.  I see no conflict here...

Reply:
Ah, that's a good point. We don't actually hard code locations (they already come from the env) so that's moot.